tools/suggest-grid.mjs

```diff
@@ -330,96 +330,200 @@ async function main({ preserve, except, apply, seed }) {
+      // let duplicates = allchairnames.filter((e, i, a) => a.indexOf(e) !== i);
+      // if (duplicates.length) {
+      // console.log("<p>Chair conflicts: " + duplicates.join(", ") + "</p>");
+      // }
+
+      // Warn if two sessions from the same track are scheduled in this slot
+      let alltracks = row.filter((s,i) => i > 0 && typeof(s) === 'object').map(s => s.tracks).flat(1);
+      let trackdups = alltracks.filter((e, i, a) => a.indexOf(e) !== i);
+      if (trackdups.length) {
+         console.log('<p class="track-error">Same track: ' + trackdups.join(", ") + "</p>");
+       }
+      console.log('</th>')
+      // Format rest of row
+      for (let i = 1; i<row.length; i++) {
+          let session = row[i];
+	  if (!session) { 
+	      console.log('<td></td>');
+	  } else {
+	      // Warn if session capacity estimate exceeds room capacity
+	      let sloterrors = [];
+	      if (session.description.capacity > rooms[i-1].capacity) {
+		  sloterrors.push('capacity-error');
+	      }
+	      if (trackdups.length && trackdups.some(r => session.tracks.includes(r))) {
+		  sloterrors.push('track-error');
+	      }
+	      if (sloterrors.length) {
+     	          console.log('<td class="' + sloterrors.join(' ') + '">');
+	      } else { 	  
+    	          console.log('<td>');
+              }
+	      let url= 'https://github.com/' + session.repository + '/issues/' + session.number;
+	      // Format session number (with link to GitHub) and name
+      	      console.log('<a href=\"' + url + '\">#' + session.number + '</a>: ' + session.title);
+	      
+	      // Format chairs
+	      console.log("<p>");
+	      console.log("<i>" + session.chairs.map(x => x.name).join(',<br/>') + "</i>");
+      	      console.log("</p>");
+	      
+	      // List session conflicts to avoid and highlight where there is a conflict.
+	      if (Array.isArray(session.description.conflicts)) {
+		  let confs = [];
+		  for (const conflict of session.description.conflicts) {
+	              for (const v of row) {
+			  if (v.number === conflict) {
+			      confs.push(conflict);
+			  }
+		      }
+		  }
+		  if (confs.length) {
+		      console.log("<p><b>Conflicts with</b>: " + confs.map(s => '<span class="conflict-error">' + s + '</span>').join(', ') + "</p>");
+		  }
+		  // This version prints all conflict info if we want that
+                  // console.log("<p><b>Conflicts</b>: " + session.description.conflicts.map(s => confs.includes(s) ? '<span class="conflict-error">' + s + '</span>' : s).join(', ') + "</p>");
+	      }
+	      if (sloterrors.includes('capacity-error')) {
+              console.log('<p><b>Capacity</b>: ' + session.description.capacity + '</p>');
+	      }
+              console.log('</td>');
+	  }
+      }	 
+      console.log('</tr>');     
+  }	
+  console.log('</table>');
+    // If any sessions have not been assigned to a room, warn us.
+  let unscheduled = sessions.filter(s => !s.slot && !s.room);
```

Comment:
Note this hould rather be a logical OR than an AND because we may have forced a room or a slot onto a session before running the script.

Reply:
Fixed in my update!